synthesizer/src/block/transaction/mod.rs

```diff
@@ -92,8 +96,13 @@ impl<N: Network> Transaction<N> {
         let deployment = vm.deploy(program, rng)?;
         // Compute the fee.
         let (_, fee, _) = vm.execute_fee(private_key, credits, fee_in_gates, query, rng)?;
+
+        // Construct the owner.
+        let id = *Self::deployment_tree(&deployment, &fee)?.root();
+        let owner = Owner::new(private_key, id.into(), rng)?;
+
         // Initialize the transaction.
-        Self::from_deployment(deployment, fee)
+        Self::from_deployment(deployment, fee, owner)
```

Comment:
Is it necessary to have this computation here? This adds a re-computation of the deployment Merkle Tree when in this function, it's assured to be computed correctly. A few cycles can be saved by just creating the deployment directly.

```suggestion
        Ok(Self::Deploy(id.into(), Box::new(deployment), fee, owner))
```

Reply:
The purpose is to guarantee corrupt dbs immediately detect failure upon deserialization.